yarn-project/simulator/src/public/enqueued_calls_processor.ts

```diff
@@ -96,13 +104,15 @@ export class EnqueuedCallsProcessor {
     globalVariables: GlobalVariables,
     historicalHeader: Header,
     worldStateDB: WorldStateDB,
+    realAvmProvingRequests: boolean = true,
```

Comment:
When this is false, I generate "empty" proving requests for the AVM, and I completely ignore the "deprecated" `PublicFunctionCallResult` that has the recursive vector of nested executions. I added this so that I can test things without caring about that deprecated type